src/main/java/DilemmaDetector/Simulator/CollisionDetector.java

```diff
@@ -1,28 +1,37 @@
 package DilemmaDetector.Simulator;
 
 import generator.Model;
+import project.Surrounding;
 
+import java.util.ArrayList;
 import java.util.LinkedList;
 import java.util.List;
 
 public class CollisionDetector {
 
     private Model scenarioModel;
     private List<Actor> actors;
+    private List<RigidBody> surroundingRigidBodies;
     private Actor mainVehicle;
 
     public CollisionDetector(Model model, Actor mainVehicle,
-                             List<Actor> actors) {
+                             List<Actor> actors, List<RigidBody> surroundingRigidBodies) {
         this.scenarioModel = model;
         this.mainVehicle = mainVehicle;
         this.actors = actors;
+        this.surroundingRigidBodies = surroundingRigidBodies;
     }
 
     public List<Actor> detectCollisionInMoment() {
         List<Actor> collidedActors = new LinkedList<>();
-        if (detectOutOfRoad(mainVehicle)) {
-            collidedActors.add(mainVehicle);
+        for (RigidBody surroundingRigidBody: surroundingRigidBodies) {
+            if (detectCollisionWithSurrounding(surroundingRigidBody)) {
+                collidedActors.add(mainVehicle);
+            }
```

Comment:
w ten sposób nie możemy ocenić z czym zderzył się pojazd, czy to był krzak, czy betonowy mur, na razie na to chyba nie patrzymy, ale czuję, że przyjdzie czas gdzie będziemy to rozróżniać, np w kontekście łamania prawa